CardMaker/Card/Translation/JavaScriptTranslator.cs

```diff
@@ -76,83 +75,58 @@ protected override ElementString TranslateToElementString(Deck zDeck, string sRa
             }
-
             foreach (var kvp in zDeckLine.ColumnsToValues)
             {
-                AddVar(zBuilder, kvp.Key, kvp.Value);
+                AddGlobal(engine, kvp.Key, kvp.Value);
             }
-            zBuilder.Append(sDefinition);
-            return zBuilder.ToString();
-        }
 
-        private void AddNumericVar(StringBuilder zBuilder, string sVar, string sValue)
-        {
-            zBuilder.Append("this.");
-            zBuilder.Append(sVar.Replace(' ', '_'));
-            zBuilder.Append("=");
-            zBuilder.Append(sValue);
-            zBuilder.AppendLine(";");
+            engine.AddHostObject("Element", zElement);
+
+            return hostFunctions;
         }
 
-        private void AddVar(StringBuilder zBuilder, string sVar, string sValue)
+        private void AddGlobal(ScriptEngine zEngine, string sName, string sValue)
         {
-            zBuilder.Append("this.");
-            zBuilder.Append(sVar.Replace(' ', '_'));
-            zBuilder.Append("=");
-            // functions or single quoted items are left as-is
-            // note this does not tolerate (whitespace)'
-            if (sValue.StartsWith(FUNCTION_PREFIX) && ProjectManager.Instance.LoadedProject.jsKeepFunctions)
-            {
-                zBuilder.AppendLine(sValue);
-            }
-            else if (sValue.StartsWith("'") && !ProjectManager.Instance.LoadedProject.jsEscapeSingleQuotes)
-            {
-                zBuilder.Append(sValue);
-                zBuilder.AppendLine(";");
-            }
-            else if (sValue.StartsWith("~") && ProjectManager.Instance.LoadedProject.jsTildeMeansCode)
+            //ProjectManager.Instance.LoadedProject.jsEscapeSingleQuotes;
+            if (ProjectManager.Instance.LoadedProject.jsKeepFunctions && sValue.StartsWith(FUNCTION_PREFIX))
             {
-                zBuilder.Append(sValue.Substring(1));
-                zBuilder.AppendLine(";");
+                AddCode(zEngine, sName, sValue);
+                return;
             }
-            else
+
+            if (ProjectManager.Instance.LoadedProject.jsTildeMeansCode)
             {
-                if (ProjectManager.Instance.LoadedProject.jsEscapeSingleQuotes)
+                if (sValue.StartsWith("\\\\~") || sValue.StartsWith("\\~"))
+                {
+                    sValue = sValue.Remove(0, 1);
+                }
+                else if (sValue.StartsWith("~"))
                 {
-                    sValue = sValue.Replace("'", @"\'");
+                    AddCode(zEngine, sName, sValue.Remove(0, 1));
+                    return;
                 }
-                zBuilder.Append("'");
-                zBuilder.Append(sValue);
-                zBuilder.AppendLine("';");
             }
+
+            zEngine.Global.SetProperty(sName, sValue);
+            if (sName.Contains(" "))
+            {
+                zEngine.Global.SetProperty(sName.Replace(" ", "_"), sValue);
```

Comment:
Probably. With this you can access a column with `column_name` or `this['column name']`. In fact, any non-javascript variable valid column name will be accessible via this[], and I thought it would be worth preserving that for spaces as well as the current replace with _ behavior. The if is just to make sure you don't add the same name twice. Perhaps unnecessary though.